crates/services/consensus_module/poa/src/ports.rs

```diff
@@ -142,3 +142,42 @@ impl PredefinedBlocks for InMemoryPredefinedBlocks {
 pub trait GetTime: Send + Sync {
     fn now(&self) -> Tai64;
 }
+
+pub trait TriggerBlockProduction: Send + Sync {
+    fn wait_for_trigger(&self) -> impl core::future::Future<Output = ()> + Send;
+}
```

Comment:
Again the terminology is a bit confusing. We're not triggering block production here, we're awaiting the dependencies to be ready.

```suggestion
pub trait WaitForReadySignal: Send + Sync {
    fn wait_for_ready_signal(&self) -> impl core::future::Future<Output = ()> + Send;
}
```

Reply:
done in [8bbfc81](https://github.com/FuelLabs/fuel-core/pull/2832/commits/8bbfc818b36d286ec04099b5b949520cf0e0b593) & [324d5a0](https://github.com/FuelLabs/fuel-core/pull/2832/commits/324d5a054538567c2889922bdede0beb392ea94a) & [20467cd](https://github.com/FuelLabs/fuel-core/pull/2832/commits/20467cd69c3dbdd3b911979add2329f1e44d66cf) & [e124d36](https://github.com/FuelLabs/fuel-core/pull/2832/commits/e124d364f3fbbcf2f26a41b3373bcde65c5503d5)

Reply:
Nice, thank you!